lua/telescope/previewers.lua

```diff
@@ -34,6 +38,38 @@ end
 
 local add_quotes = valuate_shell()
 
+local function readFile(path, callback)
+  vim.loop.fs_open(path, "r", 438, function(err, fd)
+    assert(not err, err)
+    vim.loop.fs_fstat(fd, function(err, stat)
+      assert(not err, err)
+      vim.loop.fs_read(fd, stat.size, 0, function(err, data)
+        assert(not err, err)
+        vim.loop.fs_close(fd, function(err)
+          assert(not err, err)
+          return callback(data)
+        end)
+      end)
+    end)
+  end)
+end
+
+local filetype_cache = vim.fn.split(vim.fn.execute('autocmd filetypedetect'), '\n')
+local determine_filetype = function(filepath)
```

Comment:
I would save the filepath / maybe just extension and cache the result. So you can easily do `.lua` -> `lua`